demos/object_detection_demo_yolov3_async/main.cpp

```diff
@@ -309,6 +309,8 @@ int main(int argc, char *argv[]) {
         cv::Size graphSize{frame.cols / 4, 60};
         Presenter presenter(FLAGS_u, frame.rows - graphSize.height - 10, graphSize);
 
+        cv::VideoWriter outVideo(FLAGS_o, cv::VideoWriter::fourcc('M','J','P','G'), cap->fps(), frame.size());
```

Comment:
OpenCV reports assertion fail for me if I try to open `VideoWriter` with empty path.

You should create empty `VideoWriter`. If `FLAGS_o` is not empty open it. You should also report an error if `isOpened()` returns false after `open()` because it looks that `VideoWriter` will silently accept frames to write even if it is closed.